packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -585,7 +585,7 @@ function renderWeightedTargets(weightedTargets: WeightedTarget[]): CfnRoute.Weig
   for (const t of weightedTargets) {
     renderedTargets.push({
       virtualNode: t.virtualNode.virtualNodeName,
-      weight: t.weight || 1,
+      weight: t.weight == null ? 1 : t.weight,
```

Comment:
You probably want to compare against `undefined` here instead of null.

Reply:
Done, thanks.